openff/interchange/interop/lammps/export/export.py

```diff
@@ -294,7 +294,10 @@ def _write_atoms(lmp_file: IO, interchange: Interchange, atom_type_map: dict):
         atom_map[atom_index] = atom
 
     for molecule_index, molecule in enumerate(interchange.topology.molecules):
-        molecule_hash = molecule.ordered_connection_table_hash()
+        # molecule_hash = molecule.ordered_connection_table_hash()
+        molecule_hash = f"{molecule_index}_{molecule.ordered_connection_table_hash()}"  # salt with mol ID to allow chemically-identical Molecules to be labelled with distinct IDs)
```

Comment:
IIUC, salting is the process of adding ~random data into hashes, which isn't what you're after here
```suggestion
        # inject mol ID into hash to allow chemically-identical Molecules
        # to be labelled with distinct IDs
        molecule_hash = f"{molecule_index}_{molecule.ordered_connection_table_hash()}"
```